### PR TITLE
luminous: sse-c-fixes

### PR DESCRIPTION
**Pull request by theanalyst**
http://tracker.ceph.com/issues/39070

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by cbodley** [comment]
thanks @theanalyst - i see a make check failure, but jenkins.ceph.com isn't responding to see the logs

are you able to run the new s3tests against this branch? it would be nice to verify that, since the teuthology testing won't include those

**Comment by yuriw** [comment]
[wip-yuri4-testing-2019-04-01-1924-luminous](https://trello.com/c/rBPcQR14/572-wip-yuri4-testing-2019-04-01-1924-luminous)

File: src/test/rgw/test_rgw_crypto.cc [hunk 1]
```diff
@@ -526,6 +528,172 @@ TEST(TestRGWCrypto, check_RGWGetObj_BlockDecrypt_fixup)
...
+  ASSERT_EQ(fixup_range(&decrypt, na_obj_size -2, na_obj_size -1),
+	    range_t(na_obj_size - 2, na_obj_size -1));
+
+}
+
+TEST(TestRGWCrypto, check_RGWGetObj_BlockDecrypt_fixup_invalid_ranges)
+{
+
+  ut_get_sink get_sink;
+  auto nonecrypt = std::make_unique<BlockCryptNone>(4096);
```
Comment (cbodley) [inline src/test/rgw/test_rgw_crypto.cc]: s/std::/ceph::/ here too

Comment (theanalyst) [reply src/test/rgw/test_rgw_crypto.cc]: thanks, I had actually tested the version without the last 2 commits much before, updated this and also ran sse-c s3tests against the new branch and seems to work as expected

**Comment by theanalyst** [comment]
> thanks @theanalyst - i see a make check failure, but jenkins.ceph.com isn't responding to see the logs
> 
> are you able to run the new s3tests against this branch? it would be nice to verify that, since the teuthology testing won't include those

Yeah, just verified today again, sse_c tests of the master branch with boto/boto3 seems to pass as expected 

```sh
aws) [abhishekl@d18:~/spells/storage/s3-tests](⎇ ceph-master)$
S3TEST_CONF=s3.conf ./virtualenv/bin/nosetests s3tests_boto3.functional.test_s3 -m 'sse_c' -vv
nose.config: INFO: Ignoring files matching ['^\\.', '^_', '^setup\\.py$']
s3tests_boto3.functional.test_s3.test_encryption_sse_c_method_head ... ok
s3tests_boto3.functional.test_s3.test_encryption_sse_c_present ... ok
s3tests_boto3.functional.test_s3.test_encryption_sse_c_other_key ... ok
s3tests_boto3.functional.test_s3.test_encryption_sse_c_invalid_md5 ... ok
s3tests_boto3.functional.test_s3.test_encryption_sse_c_no_md5 ... ok
s3tests_boto3.functional.test_s3.test_encryption_sse_c_no_key ... ok
s3tests_boto3.functional.test_s3.test_encryption_key_no_sse_c ... ok
s3tests_boto3.functional.test_s3.test_encryption_sse_c_multipart_upload ... ok
s3tests_boto3.functional.test_s3.test_encryption_sse_c_unaligned_multipart_upload ... ok
s3tests_boto3.functional.test_s3.test_encryption_sse_c_multipart_invalid_chunks_1 ... ok
s3tests_boto3.functional.test_s3.test_encryption_sse_c_multipart_invalid_chunks_2 ... ok
s3tests_boto3.functional.test_s3.test_encryption_sse_c_multipart_bad_download ... ok
s3tests_boto3.functional.test_s3.test_encryption_sse_c_post_object_authenticated_request ... ok

----------------------------------------------------------------------
Ran 13 tests in 72.290s

OK
(aws) [abhishekl@d18:~/spells/storage/s3-tests](⎇ ceph-master)$
S3TEST_CONF=s3.conf ./virtualenv/bin/nosetests s3tests.functional.test_s3 -m 'sse_c' -vv
nose.config: INFO: Ignoring files matching ['^\\.', '^_', '^setup\\.py$']
s3tests.functional.test_s3.test_encryption_sse_c_method_head ... ok
s3tests.functional.test_s3.test_encryption_sse_c_present ... ok
s3tests.functional.test_s3.test_encryption_sse_c_other_key ... ok
s3tests.functional.test_s3.test_encryption_sse_c_invalid_md5 ... ok
s3tests.functional.test_s3.test_encryption_sse_c_no_md5 ... ok
s3tests.functional.test_s3.test_encryption_sse_c_no_key ... ok
s3tests.functional.test_s3.test_encryption_key_no_sse_c ... ok
s3tests.functional.test_s3.test_encryption_sse_c_multipart_upload ... ok
s3tests.functional.test_s3.test_encryption_sse_c_multipart_invalid_chunks_1 ... ok
s3tests.functional.test_s3.test_encryption_sse_c_multipart_invalid_chunks_2 ... ok
s3tests.functional.test_s3.test_encryption_sse_c_multipart_bad_download ... ok
s3tests.functional.test_s3.test_encryption_sse_c_post_object_authenticated_request ... ok

----------------------------------------------------------------------
Ran 12 tests in 25.432s

OK
```

**Pull request merged**